src/ngraph/frontend/onnxifi/backend.hpp

```diff
@@ -0,0 +1,85 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>  // std::shared_ptr
+#include <string>  // std::string
+#include <utility> // std::move
+#include <vector>  // std::vector
+
+#include "ngraph/function.hpp"
+#include "ngraph/runtime/backend.hpp"
+#include "ngraph/runtime/tensor_view.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        class Backend
+        {
+        public:
+            Backend(const Backend&) = delete;
+            Backend& operator=(const Backend&) = delete;
+
+            Backend(Backend&&) = default;
+            Backend& operator=(Backend&&) = default;
+
+            Backend() = delete;
+
+            explicit Backend(std::string type)
```

Comment:
take a const string& and then skip the move statement. Exact same number of copies and more normal way to do  it.

Reply:
This change was suggested by clang-tidy, to pass by value and use move semantics.

_The purpose of `clang-tidy` is to provide an extensible framework for diagnosing and fixing typical programming errors, like style violations, interface misuse, or bugs that can be deduced via static analysis._

BTW. Why do you think move semantics is not a normal way of expressing things in C++?

Reply:
Please change it. For some reason you refuse to make your code look like all the rest of the code in the repo and seem to do things just to be different or to appease clion.